code/modules/events/nothing_happens.dm

```diff
@@ -10,3 +9,11 @@
 	min_players = 1
 	max_occurrences = INFINITE_EVENT_OCURRENCES
 	alert_observers = FALSE
+
+/datum/round_event_control/nothing_happens/New()
+	. = ..()
+	for(var/datum/round_event_control/possible_event in subtypesof(/datum/round_event_control))
+		if(possible_event.type == src.type)
+			break
+		src.weight -= possible_event.weight
```

Comment:
this won't work and it's obvious this is untested

`subtypesof()` gets a list of paths, so for loop will iterate exactly 0 times (add `as anything`)
additionally, you're trying to get a var on a path without using initial(), which is also necessary
finally, remove the `src.`

Reply:
SAD